docs/resources/task.md

```diff
@@ -281,25 +283,86 @@ spec:
 
 ## Permission Set
 
+### Generic
+
 A set of "permissions" assigned to the `task` instance, format depends on the cloud provider
 
-#### Amazon Web Services
+### Cloud-specific
+
+#### Kubernetes
+
+The name of a service account in the current namespace.
+
+### Amazon Web Services
 
 An [instance profile `arn`](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/iam-roles-for-amazon-ec2.html), e.g.:
 `permission_set = "arn:aws:iam:1234567890:instance-profile/rolename"`
 
-#### Google Cloud Platform
+### Google Cloud Platform
 
 A service account email and a [list of scopes](https://cloud.google.com/sdk/gcloud/reference/alpha/compute/instances/set-scopes#--scopes), e.g.:
 `permission_set = "sa-name@project_id.iam.gserviceaccount.com,scopes=storage-rw"`
 
-#### Microsoft Azure
+### Microsoft Azure
 A comma-separated list of [user-assigned identity](https://docs.microsoft.com/en-us/azure/active-directory/managed-identities-azure-resources/overview) ARM resource ids, e.g.:
 `permission_set = "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.ManagedIdentity/userAssignedIdentities/{identityName}"`
 
+## Pre-allocated blob container
+
+### Generic
```

Comment:
again, would remove "### Generic" here since there are no generic options for `container`, right? (for example, `machine="m"` is generic, `machine="t2.micro"` is not).